client/src/components/layout/EditForm.js

```diff
@@ -0,0 +1,190 @@
+
+  const tripId = props.match.params.id
+
+  const findTrip = async () => {
+    try {
+      const response = await fetch(`/api/v1/trips/${tripId}`)
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      }
+      const tripData = await response.json()
+      
+      const { city, continent, country, description, id, numberOfDays, title, userId } = tripData.trip
+      setForm({ city, continent, country, description, id, numberOfDays, title, userId })
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  useEffect(() => {
+    findTrip()
+  }, [])
+
+  const handleInputChange = (event) => {
+    setForm({
+      ...form,
+      [event.currentTarget.name]: event.currentTarget.value,
+    })
+  }
+
+  const editTrip = async () => {
+    try {
+      const response = await fetch(`/api/v1/trips/${tripId}`, {
+        method: "PATCH",
+        headers: new Headers({
+          "Content-Type": "application/json",
+        }),
+        body: JSON.stringify(form),
+      })
+      if (!response.ok) {
+        if (response.status == 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        }
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      } else {
+        setShouldRedirect(true)
+      }
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  if (shouldRedirect) {
+    return <Redirect to={`/${tripId}`} />
+  }
+
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    editTrip()
+  }
+
+  return (
+    <form className="form" onSubmit={handleSubmit}>
+      <ErrorList errors={errors} />
+
+      <div className="grid-container">
+        <div className="grid-x grid-padding-x">
+          <div className="medium-12 cell">
+            <label htmlFor="title">
+              {" "}
+              <span className="label">Title of the Trip</span>
+              
+              <input
+                type="text"
+                id="title"
+                name="title"
+                onChange={handleInputChange}
+                value={form.title}
+              />
+            </label>
+          </div>
+
+          <div className="medium-3 cell">
+            <label htmlFor="continent">
+              {" "}
+              <span className="label">Continent</span>
+              <input
+                type="text"
+                id="continent"
+                name="continent"
+                onChange={handleInputChange}
+                value={form.continent}
+              />
+            </label>
+          </div>
+
+          <div className="medium-3 cell">
+            <label htmlFor="country">
+              {" "}
```

Comment:
do you mind letting me know what line 123 is doing? if necessary for styling, ignore, but it does seem odd.